README.md

```diff
@@ -192,7 +191,7 @@ const getFooIdByBar = async (connection: DatabaseConnectionType, bar: string): P
 
 ```js
 const getFooIdByBar = (connection: DatabaseConnectionType, bar: string): Promise<DatabaseRecordIdType> => {
-  return connection.oneFirst(sql`
+  return connection.oneFirst<DatabaseRecordType>(sql`
```

Comment:
What's the reason all the first types ask for `DatabaseRecordType`?

This should be a primitive value, e.g.

```js
connection.oneFirst<number>(sql``);
```

Reply:
Good point. The simple reason is that this is how it works in `@types/slonik`. The slightly more complicated reason is that the generic should really be on the tag, not the function call:

```typescript
connection.oneFirst(sql<DatabaseRecordType>`select id from foo`)
```

And there it _should_ be `DatabaseRecordType`, because the template `select id from foo` represents a pg return value looking like `{ rows: [{ id: '...' }] }`. So that query's typearg should be the full record. But I'll see if it's possible to switch up the `oneFirst`, `maybeOneFirst`, `anyFirst` and `manyFirst` type args to be more intuitive anyway. Do we care about breaking changes for `@types/slonik` users?

Reply:
@gajus updated - it's now

```typescript
connection.oneFirst<number>(sql``) // resolves to type `number`
connection.maybeOneFirst<number>(sql``) // resolves to type `number | null`
connection.anyFirst<number>(sql``) // resolves to type `number[]`
connection.manyFirst<number>(sql``) // resolves to type `number[]`
```

per your suggestion.